statsmodels/base/_screening.py

```diff
@@ -0,0 +1,173 @@
+
+    pearson_resid: inconsistency in whether penalized or half-trimmed params
+    are used. Related issue: GLM resid_pearson does not include freq_weights.
+
+    freq_weights are not supported in this. Candidate ranking uses
+    moment condition with resid_pearson without freq_weights.
+
+    fit_kwds are missing, e.g. we need to avoid irls in GLM
+
+    variable names: do we keep track of those?
+
+    currently only supports numpy arrays, now exog type check or conversion
+
+    extensions to lazy or patched exog.
+    Actually, that should work already by calling screen_vars with patches
+    of exog, and then with a combined exog of best subset exog.
+
+    currently only single columns are selected, no terms (multi column exog)
+
+    """
+
+    def __init__(self, model, base_class, **kwargs):
+
+        self.model = model
+        self.model_class = model.__class__
+        self.init_kwds = model._get_init_kwds()
+        # pen_weight and penal are explicitly included
+        # TODO: check what we want to do here
+        self.init_kwds.pop('pen_weight', None)
+        self.init_kwds.pop('penal', None)
+
+        self.endog = model.endog
+        self.exog_keep = model.exog
+        self.penal = SCADSmoothed(0.1, c0=0.0001)
+
+        # this is only needed for initial start_params
+        self.base_class = base_class
+
+        # option for screening algorithm
+        self.k_add = 5
+        self.k_max_add = 10
+        self.threshold_trim = 1e-4
+        self.k_max_included = 20
+
+    def screen_vars(self, exog, endog=None, maxiter=5, disp=0):
+        model_class = self.model_class
+        if endog is None:
+            # allow a different endog than used in model
+            endog = self.endog
+        x0 = self.exog_keep
+        x1 = exog
+        k0 = x0.shape[1]
+        # TODO: remove the need for x, use x1 separately from x0
+        # needs change to idx to be based on x1 (candidate variables)
+        x = np.column_stack((x0, x1))
+        nobs, k_vars = x.shape
+
+
+        history = defaultdict(list)
+        idx_nonzero = [0]
+        keep = np.array([True])
+        idx_excl = np.arange(1, k_vars)
+        #start_params = [endog.mean()]
+        res_pen = model_class(endog, x0, **self.init_kwds).fit(disp=disp)
+        start_params = res_pen.params
+
+        for _ in range(maxiter):
+            # This does not work, droping the Poisson fit creates problems
+            p = res_pen.params.copy()
+            p[~keep] = 0
+            if hasattr(res_pen, 'resid_pearson'):
+                # this is different from the else
+                # here we use the resid_pearson from res_pen which includes
+                # dropped variables/params
+                resid_pearson = res_pen.resid_pearson
+            else:
+                predicted = res_pen.model.predict(p)
+                # this is currently hardcoded for Poisson
+                resid_pearson = (endog - predicted) / np.sqrt(predicted)
```

Comment:
should I use resid_pearson or divided by variance  (resid / variance) ?
both are orthogonality conditions but the second might correspond to WLS because exog is not weighted (i.e. is not wexog)

Reply:
Add freq_weights if in model